docs/contributing/index.md

```diff
@@ -53,18 +75,20 @@ Once you've completed these steps, you are ready to submit your contribution.
 
 ### Please format and lint as you go
 
-```bash
-make format lint
-```
 
-We attempt to provide comprehensive type annotations within this repository.  If
-you do not provide fully annotated functions or methods, the `lint` command will
-fail.  Over time, we plan to increase type-checking strictness in order to
-ensure more precise, beneficial type annotations.
+#### Usage of Pre-Commit
+
+To maintain code quality, we use pre-commit for automated checks before committing changes. Install and configure pre-commit to ensure high-quality contributions.
+
+To set up pre-commit, follow these steps:
+
+- `pip install pre-commit` for Windows and for Linux refer to your distribution's package index
+- `pre-commit install` to enable it to run automatically
+- `pre-commit run -a`  to run it manually
+
+We attempt to provide comprehensive type annotations within this repository. If you do not provide fully annotated functions or methods, the `lint` command will fail.  Over time, we plan to increase type-checking strictness in order to ensure more precise, beneficial type annotations.
```

Comment:
```suggestion
```

Since we removed the doc for the `make lint` command, this is a little confusing :)